wordpress-mobile/WordPress-FluxC-Android
# Remove sites from local DB if they aren't returned in FETCH_SITES response

cc @aforcier 

This one is a follow-up to wordpress-mobile/WordPress-Android#6052. I've tested it on that branch in WPAndroid with the latest commit from this branch as its dependency.

To test via WPAndroid:
1. Login to a .com account, head to site picker
2. Add a site to your account on the web
3. Refresh the sites list and verify the site is added
4. Delete the site from the web
5. Refresh the sites list and verify it's been removed

You'll also need to make sure local changes allow the site to stay:
1. Login to a .com account, head to site picker
2. Add a site to your account on the web
3. Refresh the sites list and verify the site is added
4. Make some local only changes to a post on the site
5. Refresh the sites list and verify it's not been removed
